### PR TITLE
[Backport]Sorting by Websites not working in product grid in backoffice

### PR DESCRIPTION
**Pull request by XxXgeoXxX**
Issue https://github.com/magento/magento2/issues/20511

#### Preconditions (*)
Magento Version : CE 2.2X
Magento Version : CE 2.3.X

#### Steps to reproduce (*)
Navigate to admin catalog product grid
Using "Columns" selection add Websites'' column in the grid
#### Expected result (*)
Sort by 'Websites' should be working fine.

#### Actual result (*)
Sort by 'websites' product attribute not working. 

### Fixed Issues (if relevant)
Backport for https://github.com/magento/magento2/pull/20512

### Manual testing scenarios (*)
Navigate to admin catalog product grid
Using "Columns" selection add Websites'' column in the grid

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @XxXgeoXxX. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.2-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by XxXgeoXxX** [comment]
@magento-engcom-team give me 2.2.2 instance

**Comment by magento-engcom-team** [comment]
Hi @XxXgeoXxX. Thank you for your request. I'm working on Magento 2.2.2 instance for you

File: app/code/Magento/Catalog/Ui/Component/Listing/Columns/Websites.php [hunk 1]
```diff
@@ -20,29 +21,42 @@ class Websites extends \Magento\Ui\Component\Listing\Columns\Column
...
      * @param StoreManagerInterface $storeManager
+     * @param Helper $resourceHelper
      * @param array $components
      * @param array $data
      */
     public function __construct(
         ContextInterface $context,
         UiComponentFactory $uiComponentFactory,
         StoreManagerInterface $storeManager,
+        Helper $resourceHelper,
```
Comment (rogyar) [inline app/code/Magento/Catalog/Ui/Component/Listing/Columns/Websites.php:53]: Please, do not change arguments order. This change goes against backward compatibility. Add the resourceHelper to the very end of the arguments set

**Review by rogyar — CHANGES_REQUESTED** [review]

**Comment by sidolov** [comment]
@XxXgeoXxX , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Comment by sidolov** [comment]
@XxXgeoXxX , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**